food-bank-app/src/components/PriorityQueueItem/PriorityQueueItem.tsx

```diff
@@ -0,0 +1,157 @@
+import { useTheme } from '@rneui/themed'
+import { useState } from 'react'
+import { Image, Text, View, TouchableOpacity, StyleSheet } from 'react-native'
+import Modal from 'react-native-modal'
+const nopictureUrl = 'https://upload.wikimedia.org/wikipedia/en/6/60/No_Picture.jpg'
+
+interface PriorityQueueItemProps {
+  title: string
+  priorityLevel: number
+  imageUrl?: string
+  modal: string
+}
+
+enum Colors {
+  '#00953b',
+  '#ddc857',
+  '#dd596e'
+}
+
+enum Priorities {
+  'BAJA',
+  'MEDIANA',
+  'ALTA'
+}
+
+const PriorityQueueItem = ({
+  title,
+  priorityLevel,
+  imageUrl = nopictureUrl,
+  modal
+}: PriorityQueueItemProps): JSX.Element => {
+  const { theme } = useTheme()
+  const [isModalVisible, setModalVisible] = useState(false)
+
+  return (
+    <View
+    style={[styles.priorityQueueItemContainer, { shadowColor: theme.colors.shadow }]}>
+      <Image
+        source={{ uri: imageUrl }}
+        style={styles.imageContainer}
+        borderRadius={10}
+        borderBottomRightRadius={0}
+        borderTopRightRadius={0}
+      />
+      <View
+        style={{
+          width: '70%',
+          borderRadius: 10
+        }}>
+        <View style={styles.itemInfoContainer}>
+          <Text style={{
+            fontSize: 20
+          }}>
+            {title}
+          </Text>
+          <TouchableOpacity onPress={() => setModalVisible(true)}>
+            <Icon
+              name='info'
+              type='foundation'
+              size={30}
+              color= '#666666'
+            />
+          </TouchableOpacity>
+          <Modal isVisible={isModalVisible} onBackdropPress={() => setModalVisible(false)}>
+            <View style={styles.modalContainer}>
+              <View style={{ height: '30%' }}>
+                <Text style={{
+                  fontSize: 20,
+                  color: theme.colors.green,
+                  marginTop: 20
+                }}>
+                  ¿Qué artículos puedes donar?
+                </Text>
+              </View>
+              <View style={{
+                flexDirection: 'column',
+                height: '70%',
+                justifyContent: 'space-between'
+              }}>
+                <Text style={{ fontSize: 16 }}>{modal}</Text>
+                <TouchableOpacity onPress={() => setModalVisible(false)}>
+                  <View style={styles.modalButton}>
+                    <Text style={{
+                      color: 'white',
+                      fontSize: 15,
+                      textAlign: 'center'
+                    }}>
+                      Entendido</Text>
+                  </View>
+                </TouchableOpacity>
+              </View>
+            </View>
+          </Modal>
+        </View>
+        <View
+          style={{
+            width: '85%',
+            alignSelf: 'center',
+            marginTop: 10
+          }}>
+          <Text
+            style={{
+              color: Colors[priorityLevel],
```

Comment:
Kind of obscure how you are indexing an `enum` as if it were an `object`, maybe not the particular right use case for an enum here. It might work, but it definitely feels weird reading it. Consider refactoring it into simply an object.